Libraries/NativeAnimation/Nodes/RCTInterpolationAnimatedNode.m

```diff
@@ -9,27 +9,72 @@
 
 #import "RCTAnimationUtils.h"
 
+NSRegularExpression *regex;
+
 @implementation RCTInterpolationAnimatedNode
 {
   __weak RCTValueAnimatedNode *_parentNode;
   NSArray<NSNumber *> *_inputRange;
   NSArray<NSNumber *> *_outputRange;
+  NSArray<NSArray<NSNumber *> *> *_outputs;
+  NSArray<NSString *> *_soutputRange;
   NSString *_extrapolateLeft;
   NSString *_extrapolateRight;
+  NSUInteger _numVals;
+  bool _hasStringOutput;
+  bool _shouldRound;
+  NSArray<NSTextCheckingResult*> *_matches;
 }
 
 - (instancetype)initWithTag:(NSNumber *)tag
                      config:(NSDictionary<NSString *, id> *)config
 {
+  if (!regex) {
```

Comment:
Could you explain the changes and add a comment here that gives the overall flow of the following code. I think it would help understand what is going on.

Reply:
I essentially just tried to translate the logic from the js implementation straight. I'll take a look at documenting this once I have some time to spare. Thanks for the review!

Reply:
@janicduplessis Added comments here: https://github.com/msand/react-native/commit/a69e8817561f098df4a9f4f2593fb1c43b19894c
Would this suffice?